File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        idx = 0
        cnt = 0
        prevValue = -inf
        
        for val in nums:
            if val != prevValue:
                nums[idx] = val
                idx += 1
                prevValue = val
                cnt = 1
            else:
                cnt += 1
                prevValue = val
                if cnt <= 2:
                    nums[idx] = val
                    idx += 1
                    
        return idx
        
        
        
        

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] A) {
        int i = 2;
        for (int j = 2; j < A.length; j++)
            if (A[j] != A[i-2])
                A[i++] = A[j];
        return i;
    }
}